tests/js/spec/redux/reducers/taskDefinition.spec.js

```diff
@@ -0,0 +1,68 @@
+import {Set} from 'immutable';
+import taskDefinition, {initialState} from 'app/redux/reducers/taskDefinition';
+import {taskDefinitionActions} from 'app/redux/actions/taskDefinition';
+
+describe('taskDefinition reducer, list protocol', () => {
+  it('has expected state after requesting getting a list', () => {
+    const requested = taskDefinition(
+      initialState,
+      taskDefinitionActions.getListRequest()
+    );
+    const expected = {
+      loading: true,
+      errorMessage: null,
+      byIds: {},
+      listViewState: {
```

Comment:
Right! I think we should have a ticket on using the shared actions and reducers for all shared resources. That would mean a lot less code and some tests we can get rid of (as they really test the same thing).

One thing that's missing in the shared reducers/actions as of now is that they don't take into account the parent/child relationship you worked on (which relates back to your other comment on my hack for showing things that don't have `isGroupHeader`).

It would be nice if that task would also merge that logic into the shared action/reducers:

Added: https://snpseq.atlassian.net/browse/CLIMS-392